src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -104,37 +90,24 @@ protected virtual Task<TResponse> RemapResponseAsync(TResponse delegatedResponse
 
             var projection = await _documentMappingService.GetProjectionAsync(documentContext, absoluteIndex, cancellationToken).ConfigureAwait(false);
 
-            return await GetDelegatedResponseAsync(request, documentContext, codeDocument, sourceText, absoluteIndex, projection, cancellationToken).ConfigureAwait(false);
-        }
-
-        protected async Task<TResponse?> GetDelegatedResponseAsync(
-            TRequest request,
-            DocumentContext documentContext,
-            RazorCodeDocument codeDocument,
-            SourceText sourceText,
-            int absoluteIndex,
-            Projection projection,
-            CancellationToken cancellationToken)
-        {
-            if (!_languageServerFeatureOptions.SingleServerSupport)
+            if (projection.LanguageKind == RazorLanguageKind.Razor)
```

Comment:
This condition needs to be removed, which probably also means `HandleInRazorAsync` should be `TryHandleInRazorAsync`

Reply:
For context IIRC the language kind only ever equals Razor when you're interacting with things like Razor directives (not HTML or C#). That's what makes handing TagHelper hovers a little difficult because you can hover over Razor or HTML and either may provide a hover result

Reply:
What about adding virtual `TryHandleAsync(request, cancellationToken)` where children can override and handle before it gets delegated out? Default implementation would return null.

Reply:
I think the name is fine to have "Razor" in it.. to me its handling Razor _scenarios_ not necessarily Razor _syntax_